frontend/src/lib/services/Nominatim/NominatimService.ts

```diff
@@ -0,0 +1,199 @@
+export const NominatimAddressSchema = z.object({
+	house_number: z.string().optional(),
+	road: z.string().optional(),
+	neighbourhood: z.string().optional(),
+	suburb: z.string().optional(),
+	city: z.string().optional(),
+	town: z.string().optional(),
+	village: z.string().optional(),
+	county: z.string().optional(),
+	state: z.string().optional(),
+	'ISO3166-2-lvl4': z.string().optional(),
+	postcode: z.string().optional(),
+	country: z.string().optional(),
+	country_code: z.string().optional()
+});
+export type NominatimAddress = z.infer<typeof NominatimAddressSchema>;
+
+export const NominatimReverseResponseSchema = z.object({
+	place_id: z.number(),
+	licence: z.string(),
+	osm_type: z.string(),
+	osm_id: z.number(),
+	lat: z.string(),
+	lon: z.string(),
+	category: z.string().optional(),
+	type: z.string().optional(),
+	place_rank: z.number(),
+	importance: z.number(),
+	addresstype: z.string(),
+	name: z.string(),
+	display_name: z.string(),
+	address: NominatimAddressSchema.optional(),
+	boundingbox: z.array(z.string()).optional()
+});
+export type NominatimReverseResponse = z.infer<typeof NominatimReverseResponseSchema>;
+
+// Minimal schema for the display_name only (backwards compatible)
+export const ReverseGeocodeResponseSchema = z.object({
+	display_name: z.string()
+});
+export type ReverseGeocodeResponse = z.infer<typeof ReverseGeocodeResponseSchema>;
+
+// Search response schema
+export const NominatimSearchResultSchema = z.object({
+	place_id: z.number(),
+	licence: z.string(),
+	osm_type: z.string(),
+	osm_id: z.number(),
+	lat: z.string(),
+	lon: z.string(),
+	class: z.string().optional(),
+	type: z.string().optional(),
+	place_rank: z.number(),
+	importance: z.number(),
+	addresstype: z.string(),
+	name: z.string(),
+	display_name: z.string(),
+	boundingbox: z.array(z.string()).optional()
+});
+export type NominatimSearchResult = z.infer<typeof NominatimSearchResultSchema>;
+
+export const NominatimSearchResponseSchema = z.array(NominatimSearchResultSchema);
+export type NominatimSearchResponse = z.infer<typeof NominatimSearchResponseSchema>;
+
+// Service interface
+export interface NominatimService {
+	reverseGeocode(lat: number, lon: number): Promise<NominatimReverseResponse>;
+	search(query: string): Promise<NominatimSearchResponse>;
+}
+
+// Configuration
+export interface NominatimConfig {
+	baseUrl?: string;
+	userAgent?: string;
+}
+
+const DEFAULT_CONFIG: Required<NominatimConfig> = {
+	baseUrl: 'https://nominatim.openstreetmap.org',
+	userAgent: 'Libre311/1.0 (https://github.com/UnityFoundation-io/Libre311)'
+};
+
+// Real implementation
+export class NominatimServiceImpl implements NominatimService {
+	private config: Required<NominatimConfig>;
+
+	constructor(config: NominatimConfig = {}) {
+		this.config = { ...DEFAULT_CONFIG, ...config };
+	}
+
+	private getBaseUrl(): string {
+		// In development, use Vite proxy to bypass CORS
+		if (import.meta.env.DEV) {
```

Comment:
It's certainly a matter of preference, but I would replace these four lines with
```js
return import.meta.env.DEV ? '/nominatim' : this.config.baseUrl;
```